Webhook Search/Pagination API Docs (#1910)
FusionAuth/fusionauth-site: site/docs/v1/tech/apis/_webhook-search-request-parameters.adoc

// parameter_prefix is either blank for parameters or "search." for body

[.api]

[field]#{parameter_prefix}description# [type]#[String]# [optional]#Optional#::

> The case-insensitive string to search for in the Webhook description or URL. This can contain wildcards using the asterisk character (`*`). If no wildcards are present, the search criteria will be interpreted as `pass:[*value*]`. If this field is omitted, no filtering will be applied to the `description` or `url` fields.

Reviewer: Seems weird to not have a URL search field.

Author: Part of the problem here is how to present the search options in the UI. We'd prefer not to have two text fields on the front-end. There are a couple of other of these search APIs that use a single request field to search multiple database fields.

The other thing is deciding how to perform the search if `description` and `url` are separate fields on the request. The current code checks whether there is a match in description `OR` URL. If we were to have separate request fields, should the result still be `OR`'d? Should it be an `AND` where both have to match? Do we need another parameter to decide which logical operator is applied?

I think that the single request parameter that checks both fields is simpler to understand and use, and I don't think having a separate field adds much value.

Reviewer 2: I think Dan is correct. This is one of the changes I made to the Search APIs prior to merging. 

The API is completely separate from how we choose to use the API in our admin UI. Think of them as separate applications. 

This means we cannot use one field `description` to search on another, this would be a surprising API behavior.

[field]#{parameter_prefix}numberOfResults# [type]#[Integer]# [optional]#Optional# [default]#defaults to `25`#::
The number of results to return from the search.

[field]#{parameter_prefix}orderBy# [type]#[String]# [optional]#Optional# [default]#defaults to `name ASC`#::
The field to order the search results as well as an order direction.
+
The possible values are:
+
 * `description` - the description of the Webhook
 * `id` - the UUID of the Webhook
 * `insertInstant` - the link:/docs/v1/tech/reference/data-types#instants[instant] when the Webhook was created
 * `url` - the Webhook URL
+

> The order direction  is optional. Possible values of the order direction are `ASC` or `DESC`. If omitted, the default sort order is `ASC`.

Reviewer: ```suggestion
The order direction is optional. Possible values of the order direction are `ASC` or `DESC`. If omitted, the default sort order is `ASC`.
```

+

> For example, to order the results by the insert instant in a descending order, use `insertInstant DESC`.

Reviewer: ```suggestion
For example, to order the results by the insert instant in descending order, use `insertInstant DESC`.
```

Author: @mooreds this phrasing is consistent with existing search docs for audit logs, event logs, entity grants, and entities. We used this phrasing on all the new search API docs.

Reviewer: okay, then ignore me. Thanks!

[field]#{parameter_prefix}startRow# [type]#[Integer]# [optional]#Optional# [default]#defaults to `0`#::
The offset into the total results. In order to paginate the results, increment this value by the [field]#numberOfResults# for subsequent requests.

[field]#{parameter_prefix}tenantId# [type]#[UUID]# [optional]#Optional#::

> Restricts the results to Webhooks associated with the given Tenant, including global Webhooks.

Reviewer: Does this mean that whatever tenantId I provide, global webhooks will be returned? If so, maybe:

```suggestion
Restricts the results to Webhooks associated with the given Tenant and all global Webhooks.
```
